cpp/tests/utilities/column_utilities.cu

```diff
@@ -33,27 +35,74 @@ void expect_column_properties_equal(cudf::column_view lhs, cudf::column_view rhs
   EXPECT_EQ(lhs.type(), rhs.type());
   EXPECT_EQ(lhs.size(), rhs.size());
   EXPECT_EQ(lhs.null_count(), rhs.null_count());
-  if (lhs.size() > 0) {
-      EXPECT_EQ(lhs.nullable(), rhs.nullable());
-  }
+  EXPECT_EQ(lhs.nullable(), rhs.nullable());
   EXPECT_EQ(lhs.has_nulls(), rhs.has_nulls());
   EXPECT_EQ(lhs.num_children(), rhs.num_children());
 }
 
-// Verify elementwise equality
+class corresponding_rows_unequal {
+public:
+  corresponding_rows_unequal(table_device_view d_lhs, table_device_view d_rhs): comp(d_lhs, d_rhs) {
+  }
+  
+  cudf::experimental::row_equality_comparator<true> comp;
+    
+  __device__ bool operator()(size_type index) {
+    return !comp(index, index);
+  }
+};
+
 void expect_columns_equal(cudf::column_view lhs, cudf::column_view rhs) {
   expect_column_properties_equal(lhs, rhs);
 
   auto d_lhs = cudf::table_device_view::create(table_view{{lhs}});
   auto d_rhs = cudf::table_device_view::create(table_view{{rhs}});
 
-  EXPECT_TRUE(
-      thrust::equal(thrust::device, thrust::make_counting_iterator(0),
-                    thrust::make_counting_iterator(lhs.size()),
-                    thrust::make_counting_iterator(0),
-                    cudf::experimental::row_equality_comparator<true>{*d_lhs, *d_rhs}));
+  thrust::device_vector<int> differences;
+  differences.reserve(lhs.size());
+
+  auto diff_iter = thrust::copy_if(thrust::device,
```

Comment:
This could be made to print _all_ the differences by doing a `gather` from `lhs` and `rhs` with `differences` as the gather_map, and then you can print the result(s) of the gather.